source/local/local.go

```diff
@@ -89,22 +87,43 @@ func (ls *localSourceHandler) CacheKey(ctx context.Context, g session.Group, ind
 }
 
 func (ls *localSourceHandler) Snapshot(ctx context.Context, g session.Group) (cache.ImmutableRef, error) {
+	sessionID := ls.src.SessionID
+	if sessionID == "" {
+		return ls.snapshotWithAnySession(ctx, g)
+	}
+
+	timeoutCtx, cancel := context.WithTimeout(ctx, 5*time.Second)
+	defer cancel()
+
+	caller, err := ls.sm.Get(timeoutCtx, sessionID, false)
+	if err != nil {
+		return ls.snapshotWithAnySession(ctx, g)
```

Comment:
Do we want to use `sm.Any` when there is a session ID, but `sm.Get` failed?

Reply:
Yes